src/coding-guidelines/expressions.rst

```diff
@@ -6,6 +6,270 @@
+
+        .. code-block:: rust
+
+            fn add(si_a: i32, si_b: i32) -> Result<i32, ArithmeticError> {
+                si_a.checked_add(si_b).ok_or(ArithmeticError::Overflow)
+            }
+
+            fn sub(a: i32, b: i32) -> Result<i32, ArithmeticError> {
+                a.checked_sub(b).ok_or(ArithmeticError::Overflow)
+            }
+
+            fn mul(a: i32, b: i32) -> Result<i32, ArithmeticError> {
+                a.checked_mul(b).ok_or(ArithmeticError::Overflow)
+            }
+
+    .. compliant_example::
+        :id: compl_ex_BgUHiRB4kc4b
+        :status: draft
+
+        Wrapping behavior must be explicitly requested. This compliant example uses wrapping functions.
+
+        .. code-block:: rust
+
+            fn add(a: i32, b: i32) -> i32 {
+                a.wrapping_add(b)
+            }
+
+            fn sub(a: i32, b: i32) -> i32 {
+                a.wrapping_sub(b)
+            }
+
+            fn mul(a: i32, b: i32) -> i32 {
+                a.wrapping_mul(b)
+            }
+
+    .. compliant_example::
+        :id: compl_ex_BhUHiRB4kc4b
+        :status: draft
+
+        Wrapping behavior call also be implemented using the ``Wrapping<T>`` type as in this compliant solution.
+        The ``Wrapping<T``> type is a ``struct`` found in the ``std::num`` module that explicitly enables two's complement wrapping arithmetic for the inner type ``T`` (which must be an integer or ``usize/isize``). 
+        The ``Wrapping<T>`` type provides a consistent way to force wrapping behavior in all build modes,
+        which is useful in specific scenarios like implementing cryptography or hash functions where wrapping arithmetic is the intended behavior.
+
+        .. code-block:: rust
+
+            use std::num::Wrapping;
+
+            fn add(si_a: Wrapping<i32>, si_b: Wrapping<i32>) -> Wrapping<i32> {
+                si_a + si_b
+            }
+
+            fn sub(si_a: Wrapping<i32>, si_b: Wrapping<i32>) -> Wrapping<i32> {
+                si_a - si_b
+            }
+
+            fn mul(si_a: Wrapping<i32>, si_b: Wrapping<i32>) -> Wrapping<i32> {
+                si_a * si_b
+            }
+
+            fn main() {    
+                let si_a = Wrapping(i32::MAX);
+                let si_b = Wrapping(i32::MAX);
+                println!("Adding {} by {} has a sum of {}", si_a, si_b, add(si_a, si_b))
+            }
+
+    .. compliant_example::
+        :id: compl_ex_BgUHiSB4kc4b
+        :status: draft
+
+        Saturation semantics means that any result that falls outside the valid range of the integer type is clamped to the maximum or minimum value instead wrapping around or resulting in an error.
+        Saturation semantics always conform to this rule because they ensure that integer operations do not result in arithmetic overflow.
+        This compliant solution uses saturating functions to provide saturation semantics for some basic arithmetic operations.
```

Comment:
```suggestion
        Saturation semantics means that instead of wrapping around or resulting in an error,
        any result that falls outside the valid range of the integer type is clamped:

        - To the maximum value, if the result were to be greater than the maximum value, or
        - To the minimum value, if the result were to be smaller than the minimum,

        Saturation semantics always conform to this rule because they ensure that integer operations do not result in arithmetic overflow. 
        This compliant solution shows how to use saturating functions to provide saturation semantics for some basic arithmetic operations.
```

Added more clarity.